src/dragen_align_pa/stages.py

```diff
@@ -0,0 +1,345 @@
+
+set +x
+gcloud secrets versions access latest --secret=illumina_cpg_workbench_api --project=cpg-common | jq -r .apiKey > key
+gcloud secrets versions access latest --secret=illumina_cpg_workbench_api --project=cpg-common | jq -r .projectID > projectID
+echo "x-api-key: $(cat key)" >> $HOME/.icav2/config.yaml
+icav2 projects enter $(cat projectID)
+set -x
+"""  # noqa: E501
+
+
+def calculate_needed_storage(
+    cram: str,
+) -> str:
+    logger.info(f'Checking blob size for {cram}')
+    storage_size: int = cpg_utils.to_path(cram).stat().st_size
+    return f'{ceil(ceil((storage_size / (1024**3)) + 3) * 1.2)}Gi'
+
+
+# No need to register this stage in Metamist I think, just ICA prep
+@stage
+class PrepareIcaForDragenAnalysis(SequencingGroupStage):
+    """Set up ICA for a single realignment run.
+
+    Creates a folder ID for the Dragen output to be written into.
+    """
+
+    def expected_outputs(self, sequencing_group: SequencingGroup) -> cpg_utils.Path:
+        sg_bucket: cpg_utils.Path = sequencing_group.dataset.prefix()
+        return sg_bucket / GCP_FOLDER_FOR_ICA_PREP / f'{sequencing_group.name}_output_fid.json'
+
+    def queue_jobs(self, sequencing_group: SequencingGroup, inputs: StageInput) -> StageOutput | None:  # noqa: ARG002
+        bucket_name: str = get_path_components_from_gcp_path(path=str(object=sequencing_group.cram))['bucket']
+        outputs: cpg_utils.Path = self.expected_outputs(sequencing_group=sequencing_group)
+
+        ica_prep_job: PythonJob = prepare_ica_for_analysis.run_ica_prep_job(
+            sequencing_group=sequencing_group,
+            output=str(outputs),
+            ica_analysis_output_folder=config_retrieve(['ica', 'data_prep', 'output_folder']),
+            api_root=ICA_REST_ENDPOINT,
+            sg_name=sequencing_group.name,
+            bucket_name=bucket_name,
+        )
+
+        return self.make_outputs(
+            target=sequencing_group,
+            data=outputs,
+            jobs=ica_prep_job,
+        )
+
+
+@stage
+class UploadDataToIca(SequencingGroupStage):
+    def expected_outputs(self, sequencing_group: SequencingGroup) -> cpg_utils.Path:
+        sg_bucket: cpg_utils.Path = sequencing_group.dataset.prefix()
+        return sg_bucket / GCP_FOLDER_FOR_ICA_PREP / f'{sequencing_group.name}_fids.json'
+
+    def queue_jobs(self, sequencing_group: SequencingGroup, inputs: StageInput) -> StageOutput | None:  # noqa: ARG002
+        output: cpg_utils.Path = self.expected_outputs(sequencing_group=sequencing_group)
+
+        upload_job: BashJob = upload_data_to_ica.upload_data_to_ica(
+            sequencing_group=sequencing_group,
+            ica_cli_setup=ICA_CLI_SETUP,
+            output=str(output),
+        )
+
+        return self.make_outputs(
+            target=sequencing_group,
+            data=output,
+            jobs=upload_job,
+        )
+
+
+@stage(
+    required_stages=[PrepareIcaForDragenAnalysis, UploadDataToIca],  # type: ignore  # noqa: PGH003
```

Comment:
```suggestion
    required_stages=[PrepareIcaForDragenAnalysis, UploadDataToIca],  # noqa: PGH003
```

I think `type: ignore` is a universal ignore, `noqa: CODE` is a specific ignore. Not sure stacking them is necessary

Reply:
Think I'll clean this up later tbh, find the correct ignore code